src/rendering/hwrenderer/scene/hw_bsp.cpp

```diff
@@ -594,6 +594,23 @@ void HWDrawInfo::RenderThings(subsector_t * sub, sector_t * sector)
 void HWDrawInfo::RenderParticles(subsector_t *sub, sector_t *front)
 {
 	SetupSprite.Clock();
+	for (uint32_t i = 0; i < sub->sprites.Size(); i++)
+	{
+		DZSprite *sp = sub->sprites[i];
+		if (!sp || sp->ObjectFlags & OF_EuthanizeMe)
+			continue;
+		if (mClipPortal)
+		{
+			int clipres = mClipPortal->ClipPoint(sp->PT.Pos);
+			if (clipres == PClip_InFront) continue;
+		}
+		if (!sp->spr)
+		{
+			HWSprite sprite;
+			sp->spr = &sprite;
```

Comment:
this is wrong. you're storing a local variable as a pointer, this will inevitably lead to memory corruption.